### PR TITLE
Replace API Server's mock data responses with database calls

### PR DESCRIPTION
**Pull request by alfiedotwtf**
Most of the current API server endpoints are now pulling from a database handle.

### CODE REVIEW

File: api-server/web-server/src/main.rs [hunk 1]
```diff
@@ -37,8 +40,13 @@ async fn main() {
     let args = ApiServerWebServerConfig::parse();
     log::info!("Command line options: {args:?}");
 
+    // TODO: generalise network configuration
```
Comment (TheQuantumPhysicist) [inline api-server/web-server/src/main.rs:43]: "generalize". We use American English in the code.

**Review by TheQuantumPhysicist — APPROVED** [review]
Actually this is looking good as is. I'm happy to merge it.

File: api-server/web-server/src/main.rs (hunk 1 above)
Comment (alfiedotwtf) [reply api-server/web-server/src/main.rs:43]: Muscle memory... I'm sure there's an Emacs macro to fix my spelling 😄

File: api-server/web-server/src/api/v1.rs [hunk 2]
```diff
@@ -68,115 +81,242 @@ pub fn routes() -> Router<APIServerWebServerState> {
...
+            .into();
+
+        state
+            .db
+            .transaction_ro()
+            .map_err(|_| {
+                APIServerWebServerError::ServerError(
+                    APIServerWebServerServerError::InternalServerError,
+                )
+            })?
```
Comment (iljakuklic) [inline api-server/web-server/src/api/v1.rs:102]: Could you define a constant or something for the error? I don't like how a mere conversion of error type takes up 5 lines while the actual "substantive" code is a fraction of that.

**Review by iljakuklic — APPROVED** [review]
Looks good. A tiny style comment below:

File: api-server/web-server/src/api/v1.rs (hunk 2 above)
Comment (iljakuklic) [reply api-server/web-server/src/api/v1.rs:102]: This is replicated many times over.

Comment (alfiedotwtf) [reply api-server/web-server/src/api/v1.rs:102]: Yep. There's currently not that many error types, but once it's fully fleshed out, I'll replace the errors with real constants.

**Pull request merged**